src/deepqmc/wf/nn_wave_function/nn_wave_function.py

```diff
@@ -82,21 +83,17 @@ def __init__(
         cusp_electrons,
         cusp_alpha,
         backflow_transform,
+        conf_coeff,
     ):
         super().__init__(hamil.mol)
         n_up, n_down = self.n_up, self.n_down
         self.n_det = n_determinants
         self.full_determinant = full_determinant
-        self.envelope = envelope(hamil.mol)
-        self.mo_coeff = hk.Linear(
-            n_determinants * (n_up + n_down),
-            with_bias=False,
-            w_init=lambda s, d: hk.initializers.VarianceScaling(1.0)(s, d)
-            + jnp.ones(s),
-            name='mo_coeff',
-        )
-        self.conf_coeff = hk.Linear(
-            1, with_bias=False, w_init=jnp.ones, name='conf_coeff'
+        self.envelope = envelope(hamil.mol, n_determinants)
+        self.conf_coeff = (
```

Comment:
Having `conf_coeff` is not strictly necessary, since they can be included in the backflow. FermiNet doesn't have `conf_coeff`. We'll have to reconcile this with the excited states code.

Reply:
I agree that having `conf_coeff` is merely a redundant parametrization, since we have removed the `conf` argument to the wave function ansatz. However, I would keep it, as it can be useful for further development and hence like your solution that allows for both. I was thinking that we could maybe simplify the API further by introducing a no op and a stop graph replacement for MLPs. This logic could be applied here aswell.

Reply:
The problem is that if we don't have `conf_coeff`, the operation is not noop but `sum` (along the determinant axis). We could implement this like:
```
conf_coeff:
  _target_: jnp.sum
  _partial_: true
  keepdim: true
```
In the code we could then always write:
```
psi = self.conf_coeff(xs).squeeze()
```
What do you think? Why would we need a stop graph subnet?

Reply:
Yes, that would be a simple and clean solution. Looks good to me. The stop graph subnet would be useful to remove the convolution kwarg from the GNN, but that's a different story.

Reply:
Resolved in https://github.com/deepqmc/deepqmc/pull/121/commits/18845e56a67e3f7828c346c4ccaf25d3318e72bb

I've introduced the `SumPool` layer which is a `jnp.sum` but has the interface of a haiku module.